### PR TITLE
Realm settings(themes): Add themes tab + tests

### PR DESCRIPTION
**Pull request by jenny-s51**
## Motivation

https://issues.redhat.com/browse/APPDUX-939

## Brief Description

Adds themes tab, cypress tests for Realm Settings

## Verification Steps

1. Go to Realm Settings -> Themes tab.
2. Populate fields and save.
3. Navigate to another location in Keycloak and return to Realm Settings -> Themes tab.
4. Verify that the data saves correctly.

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] User-visible strings are using the react-i18next framework (useTranslation)
- [x] Help has been implemented
- [ ] axe report has been run and resulting a11y issues have been resolved
- [x] Unit tests have been created/updated
- [x] Formatting has been performed via prettier/eslint
- [x] Type checking has been performed via 'yarn check-types'

### CODE REVIEW

**Review by ssilvert — CHANGES_REQUESTED** [review]
You are getting the wrong values for the theme dropdowns.  Each theme type (Login, Account, Admin Console, Email) has its own set of themes.  It looks like you are using the same values for all of them.

For instance `keycloak.v2` will only show up for the account theme.

**Review by edewit — CHANGES_REQUESTED** [review]

File: src/realm-settings/ThemesTab.tsx [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+  const [defaultLocaleOpen, setDefaultLocaleOpen] = useState(false);
+
+  const [
+    internationalizationEnabled,
+    setInternationalizationEnabled,
+  ] = useState(false);
+
+  const form = useForm();
+
+  const themeTypes = ["base", "keycloak", "keycloakV2"];
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply src/realm-settings/ThemesTab.tsx]: nice catch - thank you

File: src/realm-settings/ThemesTab.tsx [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+    "nl",
+    "no",
+    "pl",
+    "ptBR",
+    "ru",
+    "sk",
+    "sv",
+    "tr",
+    "zhCN",
+  ];
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply src/realm-settings/ThemesTab.tsx]: sounds good, working on this

File: src/realm-settings/messages.json [hunk 3]
```diff
@@ -37,10 +38,45 @@
...
+      "nl": "Nederlands",
+      "no": "Norsk",
+      "pl": "Polski",
+      "pt-BR": "Português (Brasil)",
+      "ru": "Русский",
+      "sk": "Slovenčina",
+      "sv": "Svenska",
+      "tr": "Türkçe",
+      "zh-CN": "中文"
+    },
```
Comment (ssilvert) [inline src/realm-settings/messages.json:68]: This stuff is not needed, right?

**Review by ssilvert — CHANGES_REQUESTED** [review]
@jenny-s51  Everything is working well now.  Just one comment on the code.

File: src/realm-settings/messages.json (hunk 3 above)
Comment (jenny-s51) [reply src/realm-settings/messages.json:68]: We do need this - it translates the ISO codes to the language of the locale as in the mock: https://marvelapp.com/prototype/6fd5iga/screen/75632704. 

They get localized/translated here:
https://github.com/keycloak/keycloak-admin-ui/pull/544/files#diff-aa0d9341b1e4e3e2c1857217bc2d047d4ba69d985f5da6dc670c2db755cdc062R351

I tried doing this with [the `Intl.DisplayNames()` constructor ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames/DisplayNames) as @edewit suggested but it looks like TS does not recognize the `DisplayNames` prop on `Intl` without an external package/polyfill.

Comment (edewit) [reply src/realm-settings/messages.json:68]: Yeah there is no safari support for that yet, that's why it's not part of typescript. @ssilvert we also have this in the old console.

Comment (ssilvert) [reply src/realm-settings/messages.json:68]: @edewit @jenny-s51 OK.  That makes sense for the locales.  We can add a package later if needed.

But I still question lines 42-46.  What is the reason for hard-coded theme names?  We get those from serverinfo, right?

Comment (edewit) [reply src/realm-settings/messages.json:68]: @ssilvert right we don't need those

Comment (jenny-s51) [reply src/realm-settings/messages.json]: 👍 removed theme types @ssilvert @edewit

**Review by edewit — APPROVED** [review]

**Review by ssilvert — APPROVED** [review]

**Pull request merged**